pkg/bb/bb.go

```diff
@@ -289,23 +293,26 @@ func SrcFiles(p *build.Package) []string {
 // RewritePackage rewrites pkgPath to be bb-mode compatible, where destDir is
 // the file system destination of the written files and bbImportPath is the Go
 // import path of the bb package to register with.
-func RewritePackage(env golang.Environ, pkgPath, bbImportPath string, importer types.Importer) error {
+func RewritePackage(env golang.Environ, pkgPath, bbImportPath string, importer types.Importer) (bool, error) {
 	buildp, err := env.Package(pkgPath)
 	if err != nil {
-		return err
+		return false, err
 	}
 
 	p, err := NewPackage(filepath.Base(buildp.Dir), buildp.ImportPath, SrcFiles(buildp), importer)
 	if err != nil {
-		return err
+		return false, err
+	}
+	if p == nil {
+		return false, nil
 	}
 	dest := filepath.Join(buildp.Dir, ".bb")
 	// If .bb directory already exists, delete it. This will prevent stale
 	// files from being included in the build.
 	if err := os.RemoveAll(dest); err != nil {
-		return fmt.Errorf("error removing stale directory %q: %v", dest, err)
+		return false, fmt.Errorf("error removing stale directory %q: %v", dest, err)
 	}
-	return p.Rewrite(dest, bbImportPath)
+	return true, p.Rewrite(dest, bbImportPath)
```

Comment:
Did a checkout, and I was able to repro those outcome pasted above. But after I deleted `u-root` cmd, and re-build it, it gives me following issues.

```
xuehaohu@worldpeace:~/go/src/github.com/u-root/u-root$ gh pr checkout 2107
Switched to branch 'fixplan9'
xuehaohu@worldpeace:~/go/src/github.com/u-root/u-root$ ls
AUTHORS  cmds     CONTRIBUTING.md  examples  go.mod  integration  pkg        RELEASES    templates.go  tricksandtips.md  uroot_test.go
bb       configs  docs             files     go.sum  LICENSE      README.md  roadmap.md  tools         u-root.go         vendor
xuehaohu@worldpeace:~/go/src/github.com/u-root/u-root$ go build .
xuehaohu@worldpeace:~/go/src/github.com/u-root/u-root$ ls
AUTHORS  cmds     CONTRIBUTING.md  examples  go.mod  integration  pkg        RELEASES    templates.go  tricksandtips.md  u-root.go      vendor
bb       configs  docs             files     go.sum  LICENSE      README.md  roadmap.md  tools         u-root            uroot_test.go
xuehaohu@worldpeace:~/go/src/github.com/u-root/u-root$ GOOS=plan9 ./u-root plan9 cmds/core/lddfiles/
2021/08/28 11:57:01 Disabling CGO for u-root...
2021/08/28 11:57:01 Build environment: GOARCH=amd64 GOOS=plan9 GOROOT=/usr/lib/google-golang GOPATH=/usr/local/google/home/xuehaohu/go CGO_ENABLED=0
2021/08/28 11:57:01 GOOS is not linux. Did you mean to set GOOS=linux?
2021/08/28 11:57:01 WARNING: You are not using one of the recommended Go versions (have = devel, recommended = [go1.13 go1.14 go1.15 go1.16]).
			Some packages may not compile.
			Go to https://golang.org/doc/install to find out how to install a newer version of Go,
			or use https://godoc.org/golang.org/dl/go1.13 to install an additional version of Go.
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/blkid": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/blkid
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/df": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/df
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/fusermount": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/fusermount
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/insmod": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/insmod
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/ip": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/ip
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/kexec": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/kexec
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/lddfiles": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/lddfiles
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/lockmsrs": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/lockmsrs
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/losetup": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/losetup
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/lsdrivers": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/lsdrivers
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/lsmod": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/lsmod
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/msr": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/msr
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/ntpdate": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/ntpdate
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/pci": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/pci
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/rmmod": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/rmmod
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/rsdp": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/rsdp
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/sluinit": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/sluinit
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/strace": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/strace
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/stty": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/stty
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/switch_root": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/switch_root
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/sync": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/sync
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/umount": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/umount
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/uname": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/uname
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/unshare": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/unshare
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/watchdog": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/watchdog
2021/08/28 11:57:01 Skipping package "/usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/watchdogd": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/watchdogd
2021/08/28 11:57:01 Skipping package "cmds/core/lddfiles": no buildable Go source files in /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/lddfiles
2021/08/28 11:57:08 error building: type checking failed: /usr/local/google/home/xuehaohu/go/src/github.com/u-root/u-root/cmds/core/backoff/main.go:25:2: could not import flag (type-checking package "flag" failed (/usr/lib/google-golang/src/flag/flag.go:71:2: could not import errors (type-checking package "errors" failed (/usr/lib/google-golang/src/errors/wrap.go:8:2: could not import internal/reflectlite (type-checking package "internal/reflectlite" failed (/usr/lib/google-golang/src/internal/reflectlite/value.go:9:2: could not import runtime (type-checking package "runtime" failed (/usr/lib/google-golang/src/runtime/os_plan9.go:432:6: exit redeclared in this block))))))))
xuehaohu@worldpeace:~/go/src/github.com/u-root/u-root$
```

maybe `u-root`, instead of `./u-root`, picks up u-root from bin path than from you just built from current branch ? -- however, I don't see `u-root.go` depends on `pkg/bb/bb.go` -- so probably does not matter

Reply:
(maybe I build it in wrong way ?)

Reply:
don't build with google internal anything. It never ends well. Try building outside google for example.

Reply:
it does builds for me on mac :)  -- thought I am still not sure how the continue code branch is hit